Lib/asyncio/base_subprocess.py

```diff
@@ -234,10 +229,13 @@ async def _wait(self):
         self._exit_waiters.append(waiter)
         return await waiter
 
-    def _try_finish(self):
+    def _try_finish(self, force=False):
         assert not self._finished
         if self._returncode is None:
             return
+        if force:
```

Comment:
Why not always close all pipes if the process completed (if returncode is not None)?